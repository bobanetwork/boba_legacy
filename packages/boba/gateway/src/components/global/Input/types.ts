import { ReactNode } from 'react'

export interface InputProps {
  placeholder: string
  type: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: any
  label?: ReactNode
}
