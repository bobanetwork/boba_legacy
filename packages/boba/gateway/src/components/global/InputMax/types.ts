export interface MaxInputTypes {
  max: number
  initialValue?: number
  onValueChange: (value: number) => void
}
