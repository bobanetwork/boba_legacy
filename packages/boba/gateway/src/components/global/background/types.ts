export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
