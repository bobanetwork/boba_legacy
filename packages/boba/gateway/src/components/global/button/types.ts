import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
