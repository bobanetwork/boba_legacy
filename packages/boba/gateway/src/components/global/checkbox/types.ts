export interface CheckboxWithLabelProps {
  label?: string
  checked?: boolean
  onChange: (isChecked: boolean) => void
}
