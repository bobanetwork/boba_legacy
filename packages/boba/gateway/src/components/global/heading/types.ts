import { ReactNode } from 'react'

export type VariantType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

export interface HeadingProps {
  variant: VariantType
  color?: string
  children?: ReactNode
  className?: string
}

export interface HeadingStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
