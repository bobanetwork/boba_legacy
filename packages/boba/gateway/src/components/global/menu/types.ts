import { ComponentType, ReactElement, ReactNode } from 'react'

export interface OptionProps {
  label?: string
  component?: ReactElement<any>
  onClick?: (d: any) => void
}

export interface MenuProps {
  name?: string
  label?: ReactNode | string
  children?: ReactNode
  options: Array<OptionProps>
}
