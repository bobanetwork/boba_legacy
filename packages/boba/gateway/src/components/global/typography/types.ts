import { ReactNode } from 'react'

export type VariantType =
  | 'h1'
  | 'head'
  | 'title'
  | 'body1'
  | 'body2'
  | 'body3'
  | 'subtitle'

export type ComponentType = 'p' | 'span' | 'a'

export interface TypographyProps {
  variant: VariantType
  color?: string
  error?: boolean
  children?: ReactNode
  className?: string
  component?: ComponentType
}

export interface TypographyStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
