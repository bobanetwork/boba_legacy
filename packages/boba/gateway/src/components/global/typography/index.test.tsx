import { render } from '@testing-library/react'
import { TypographyProps } from './types'
import CustomThemeProvider from 'themes'
import { Typography } from '.'
import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const renderTypography = (props: TypographyProps) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
      })}
    >
      <CustomThemeProvider>
        <Typography {...props}>Text goes here</Typography>
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Typography', () => {
  test('should match snapshots h1', () => {
    const { asFragment } = renderTypography({
      variant: 'h1',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots head', () => {
    const { asFragment } = renderTypography({
      variant: 'head',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots title', () => {
    const { asFragment } = renderTypography({
      variant: 'title',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots body1', () => {
    const { asFragment } = renderTypography({
      variant: 'body1',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots body2', () => {
    const { asFragment } = renderTypography({
      variant: 'body2',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots body3', () => {
    const { asFragment } = renderTypography({
      variant: 'body3',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots subtitle', () => {
    const { asFragment } = renderTypography({
      variant: 'subtitle',
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
