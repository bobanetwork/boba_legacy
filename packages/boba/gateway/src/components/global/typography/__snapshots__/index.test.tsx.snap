// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should match snapshots body1 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB ciRvxv"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots body2 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB eCuYeS"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots body3 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB eBLtca"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots h1 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB fLFrDT"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots head 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB cgRXLP"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots subtitle 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB bqiEPL"
  >
    Text goes here
  </p>
</DocumentFragment>
`;

exports[`Typography should match snapshots title 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB goFeVO"
  >
    Text goes here
  </p>
</DocumentFragment>
`;
